ckanext/dbquery/actions/dbquery.py

```diff
@@ -43,3 +53,37 @@ def query_database(context, data_dict):
     }
 
     return resp
+
+
+@toolkit.side_effect_free
+def dbquery_executed_list(context, data_dict):
+    """
+    Return a list of all executed queries ordered by timestamp descending
+    """
+    # Check if user is authorized
+    toolkit.check_access('query_database', context, data_dict)
+
+    # Get filter parameters
+    user_filter = data_dict.get('user')
+    date_filter = data_dict.get('date')
+    limit = data_dict.get('limit')
+
+    # Get all executed queries
+    queries = DBQueryExecuted.get_all_ordered_by_timestamp_desc()
+
+    # Apply filters if provided
+    if user_filter:
+        queries = [q for q in queries if user_filter.lower() in q.user_id.lower()]
+
+    if date_filter:
+        date_obj = datetime.datetime.strptime(date_filter, '%Y-%m-%d').date()
+        queries = [q for q in queries if q.timestamp.date() == date_obj]
+
+    # Apply limit if provided
+    if limit and isinstance(limit, int):
+        queries = queries[:limit]
```

Comment:
All filter and limit should be user with the DB query at `get_all_ordered_by_timestamp_desc`
This function should be renamed to something like `get_queries(user, date, limit=10)`
For this first version it's OK to remove all filters and limits.
But if we are going to use it, it'll be done in the DB query function, not here